lib/post_destroyer.rb

```diff
@@ -66,7 +66,7 @@ def destroy
 
     delete_removed_posts_after = @opts[:delete_removed_posts_after] || SiteSetting.delete_removed_posts_after
 
-    if @user.staff? || delete_removed_posts_after < 1 || post_is_reviewable?
+    if @user.staff? || delete_removed_posts_after < 1 || post_is_reviewable? || Guardian.new(@user).is_category_group_moderator?(topic.category)
```

Comment:
I know it is slightly less efficient but we could change to:

`if delete_removed_posts_after < 1 || post_is_reviewable? || Guardian.new(@user).can_moderate?(topic)`

maybe even extract guardian somehow so we can do: 


`if delete_removed_posts_after < 1 || post_is_reviewable? || guardian.can_moderate?(topic)`